Source/FieldSolver/FiniteDifferenceSolver/MacroscopicEvolveHM_2nd.cpp

```diff
@@ -651,32 +651,62 @@ void FiniteDifferenceSolver::MacroscopicEvolveHMCartesian_2nd(
             Box const &tbx = mfi.tilebox(Hfield[0]->ixType().toIntVect());
             Box const &tby = mfi.tilebox(Hfield[1]->ixType().toIntVect());
             Box const &tbz = mfi.tilebox(Hfield[2]->ixType().toIntVect());
+            
+            // read in Ms to decide if the grid is magnetic or not
+            auto& mag_Ms_mf = macroscopic_properties->getmag_Ms_mf();
+            Array4<Real> const& mag_Ms_arr = mag_Ms_mf.array(mfi);
+            //if (mag_Ms_mf.min(1,0) == 0){
+                auto& mu_mf = macroscopic_properties->getmu_mf();
+                Array4<Real> const& mu_arr = mu_mf.array(mfi);
+            //}
```

Comment:
if the `if` condition is turned on, the following error is reported when compiling:
`./Source/FieldSolver/FiniteDifferenceSolver/MacroscopicEvolveHM_2nd.cpp:671:113: error: ‘mu_arr’ was not declared in this scope
                         Real mu_arrx = MacroscopicProperties::macro_avg_to_face(i, j, k, amrex::IntVect(1,0,0), mu_arr);`

Reply:
We cannot define a new variable inside an if() {} condition and use the varialbe outside the if(){} condition.

It is okay to define a variable withing if (){ } and use it within that same scope. 

Otherwise the variable goes out of scope and will not compile.

Suggestion is to define `mu_mf` and `mu_arr` without the if condition and call it using the if condition as is done now.

Reply:
Agreed.  The lines previously inside the if statement are just setting up pointers to data, so it's not like it's doing any "work" anyway.